test/unit/argument_parser/format_parse_validators_test.cpp

```diff
@@ -550,55 +552,55 @@ TEST(validator_test, arithmetic_range_validator_error)
     std::vector<int> option_vector;
 
     // option - above max
-    const char * argv[] = {"./argument_parser_test", "-i", "30"};
-    argument_parser parser{"test_parser", 3, argv};
+    const char * argv[] = {"./test", "--version-check", "0", "-i", "30"};
+    argument_parser parser{"test_parser", 5, argv};
     parser.add_option(option_value, 'i', "int-option", "desc",
                       option_spec::DEFAULT, arithmetic_range_validator{1, 20});
 
     EXPECT_THROW(parser.parse(), validation_failed);
 
     // option - below min
-    const char * argv2[] = {"./argument_parser_test", "-i", "-21"};
-    argument_parser parser2{"test_parser", 3, argv2};
+    const char * argv2[] = {"./test", "--version-check", "0", "-i", "-21"};
+    argument_parser parser2{"test_parser", 5, argv2};
     parser2.add_option(option_value, 'i', "int-option", "desc",
                        option_spec::DEFAULT, arithmetic_range_validator{-20, 20});
 
     EXPECT_THROW(parser2.parse(), validation_failed);
 
     // positional option - above max
-    const char * argv3[] = {"./argument_parser_test", "30"};
-    argument_parser parser3{"test_parser", 2, argv3};
+    const char * argv3[] = {"./test", "--version-check", "0", "30"};
+    argument_parser parser3{"test_parser", 4, argv3};
     parser3.add_positional_option(option_value, "desc", arithmetic_range_validator{1, 20});
 
     EXPECT_THROW(parser3.parse(), validation_failed);
 
     // positional option - below min
-    const char * argv4[] = {"./argument_parser_test", "--", "-21"};
-    argument_parser parser4{"test_parser", 3, argv4};
+    const char * argv4[] = {"./test", "--version-check", "0", "--", "-21"};
+    argument_parser parser4{"test_parser", 5, argv4};
     parser4.add_positional_option(option_value, "desc", arithmetic_range_validator{-20, 20});
 
     EXPECT_THROW(parser4.parse(), validation_failed);
 
     // option - vector
-    const char * argv5[] = {"./argument_parser_test", "-i", "-100"};
-    argument_parser parser5{"test_parser", 3, argv5};
+    const char * argv5[] = {"./test", "--version-check", "0", "-i", "-100"};
+    argument_parser parser5{"test_parser", 5, argv5};
     parser5.add_option(option_vector, 'i', "int-option", "desc",
                        option_spec::DEFAULT, arithmetic_range_validator{-50, 50});
 
     EXPECT_THROW(parser5.parse(), validation_failed);
 
     // positional option - vector
     option_vector.clear();
-    const char * argv6[] = {"./argument_parser_test", "--", "-10", "100"};
-    argument_parser parser6{"test_parser", 4, argv6};
+    const char * argv6[] = {"./test", "--version-check", "0", "--", "-10", "100"};
+    argument_parser parser6{"test_parser", 6, argv6};
```

Comment:
Is not easier to just pass true/false to the parser on construction than overwriting the arguments?